src/engine/dataapi/DataStore.java

```diff
@@ -1,17 +1,75 @@
 package engine.dataapi;
 
+import java.util.ArrayList;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.ObjectInputStream;
+import java.io.ObjectOutputStream;
+
 /**
  * Receiving end of the data storage API.
  */
-public class DataStore implements ProtoDataStore{
+public class DataStore implements ProtoDataStore {
+	private DataStream data;
 
+	
     /**
-     * Receives a data request and processes it.
+     * Receives a data request for output and processes it.
+     * we're gonna pass an id as well that we'll make in the controller to have a reference to our obj in the db
+     * @param newData is the data to be appended and/or stored
+     * @param id is the id of the object/list
+     * @param file is the user-specified file
      */
     @Override
-    public DataStream receiveDataRequest(DataStream data) {
-        return null;
+    public DataStream receiveDataRequest(DataStream newData, String id, File file) {
+        // search db or file for id
```

Comment:
We're not rrying serialize things yet, and this adds some over complication as we're not using a raw string anymore which kind of breaks the system.